homeassistant/components/moon/sensor.py

```diff
@@ -46,54 +49,40 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Moon sensor."""
-    name = config.get(CONF_NAME)
+    name: str = config[CONF_NAME]
 
     async_add_entities([MoonSensor(name)], True)
 
 
 class MoonSensor(SensorEntity):
     """Representation of a Moon sensor."""
 
-    def __init__(self, name):
+    _attr_device_class = "moon__phase"
+
+    def __init__(self, name: str) -> None:
         """Initialize the moon sensor."""
-        self._name = name
-        self._state = None
-
-    @property
-    def name(self):
-        """Return the name of the entity."""
-        return self._name
-
-    @property
-    def device_class(self):
-        """Return the device class of the entity."""
-        return "moon__phase"
-
-    @property
-    def native_value(self):
-        """Return the state of the device."""
-        if self._state < 0.5 or self._state > 27.5:
-            return STATE_NEW_MOON
-        if self._state < 6.5:
-            return STATE_WAXING_CRESCENT
-        if self._state < 7.5:
-            return STATE_FIRST_QUARTER
-        if self._state < 13.5:
-            return STATE_WAXING_GIBBOUS
-        if self._state < 14.5:
-            return STATE_FULL_MOON
-        if self._state < 20.5:
-            return STATE_WANING_GIBBOUS
-        if self._state < 21.5:
-            return STATE_LAST_QUARTER
-        return STATE_WANING_CRESCENT
-
-    @property
-    def icon(self):
-        """Icon to use in the frontend, if any."""
-        return MOON_ICONS.get(self.state)
+        self._attr_name = name
 
     async def async_update(self):
         """Get the time and updates the states."""
         today = dt_util.as_local(dt_util.utcnow()).date()
-        self._state = moon.phase(today)
+        _state = moon.phase(today)
```

Comment:
```suggestion
        state = moon.phase(today)
```
Since this is local now, it would be better to name it without `_`